test/finance/VestingWallet.test.js

```diff
@@ -0,0 +1,118 @@
+const { constants, expectEvent, expectRevert, time } = require('@openzeppelin/test-helpers');
+const { expect } = require('chai');
+
+const ERC20VotesMock = artifacts.require('ERC20VotesMock');
+const VestingWallet = artifacts.require('VestingWallet');
+
+const min = (...args) => args.slice(1).reduce((x, y) => x.lt(y) ? x : y, args[0]);
+
+contract('VestingWallet', function (accounts) {
+  const [ beneficiary, other ] = accounts;
+
+  const amount = web3.utils.toBN(web3.utils.toWei('100'));
+  const duration = web3.utils.toBN(4 * 365 * 86400); // 4 years
+
+  beforeEach(async function () {
+    this.start = (await time.latest()).addn(3600); // in 1 hour
+    this.token = await ERC20VotesMock.new('Name', 'Symbol');
+    this.vesting = await VestingWallet.new(beneficiary, this.start, duration);
+    await this.token.mint(this.vesting.address, amount);
+
+    this.schedule = Array(256).fill()
+      .map((_, i) => web3.utils.toBN(i).mul(duration).divn(224).add(this.start))
+      .map(timestamp => ({
+        timestamp,
+        vested: min(amount.mul(timestamp.sub(this.start)).div(duration), amount),
```

Comment:
```suggestion
        vested: min(amount, amount.mul(timestamp.sub(this.start)).div(duration)),
```

I think this is easier to read.